### PR TITLE
Fix #1371 error via adding errors.

### LINKED ISSUE #1371: Add checks to moderation commands to prevent banning the bot
We should prevent the bot from being able to infract itself!

Thanks to @decorator-factory for... discovering this one.

We need to add checks on moderation commands to check if the target user is the current bot.

Sentry Issue: [BOT-KS](https://sentry.io/organizations/python-discord/issues/2159129521/?referrer=github_integration)

```
AttributeError: 'ClientUser' object has no attribute 'create_dm'
(5 additional frame(s) were not displayed)
...
  File "bot/exts/moderation/infraction/_utils.py", line 179, in notify_infraction
    return await send_private_embed(user, embed)
  File "bot/exts/moderation/infraction/_utils.py", line 208, in send_private_embed
    await user.send(embed=embed)
  File "discord/abc.py", line 883, in send
    channel = await self._get_channel()
  File "discord/member.py", line 257, in _get_channel
    ch = await self.create_dm()
  File "discord/member.py", line 110, in general
    return getattr(self._user, x)(*args, **kwargs)

Error executing command invoked by fix error#1024: !ban <@!409107086526644234>
```

### PR DESCRIPTION
**Pull request by Senjan21**
This PR aims to fix #1371 via raising a new error if user being infracted is not a bot and handling it within error handler.

### CODE REVIEW

File: bot/exts/backend/error_handler.py [hunk 1]
```diff
@@ -82,6 +82,8 @@ async def on_command_error(self, ctx: Context, e: errors.CommandError) -> None:
             elif isinstance(e.original, BrandingError):
                 await ctx.send(embed=self._get_error_embed(random.choice(ERROR_REPLIES), str(e.original)))
                 return
+            elif isinstance(e.original, InvalidInfractedUser):
+                await ctx.send(f"Cannot infract that user. {e.original.reason}")
```
Comment (Akarys42) [inline bot/exts/backend/error_handler.py:86]: Getting
```py
bot_1       | Traceback (most recent call last):
bot_1       |   File "/usr/local/lib/python3.8/site-packages/discord/client.py", line 343, in _run_event
bot_1       |     await coro(*args, **kwargs)
bot_1       |   File "/bot/bot/exts/backend/error_handler.py", line 86, in on_command_error
bot_1       |     await ctx.send(f"Cannot infract that user. {e.original.reason}")
bot_1       | AttributeError: 'InvalidInfractedUser' object has no attribute 'reason'
```

**Review by Akarys42 — CHANGES_REQUESTED** [review]
Doesn't work, sir

File: bot/exts/backend/error_handler.py (hunk 1 above)
Comment (Senjan21) [reply bot/exts/backend/error_handler.py:86]: Hello, this error is no more.

**Review by kosayoda — APPROVED** [review]
:+1: Tested locally. Error reply could use a :x:, but it isn't important.

**Review by Akarys42 — APPROVED** [review]

**Pull request merged**